### ISSUE TITLE
[FR] smart `Dimension Tool`

### ISSUE THREAD

**Issue by discip** [realthunder/FreeCAD#578]
@realthunder
Here comes one more suggestion:

### Combine `Horizontal Distance`, `Vertical Distance`, `Arc`... into one _smart_ ***'Dimension Tool'*** that depending on your selections (e.g. vertex & line) and the angle to the respective selections proposes either a 'Horizontal Distance ', an 'Angle'...

Thanks in advance
kind regards

**Comment by snelweg** [comment]
yes! That is the only decent thing I found when using SolidWorks for makers!
